frontend/ui/trapper.lua

```diff
@@ -45,69 +52,74 @@ function Trapper:wrap(func)
         -- any code, but let's do it anyway.
     end
-    local co = coroutine.create(func)
+    local co = coroutine.create(pcalled_func)
     return coroutine.resume(co)
 end
 
 --- Is the code wrapped by Trapper?
--- @treturn bool true if code is wrapped by Trapper, false otherwise
+--
+-- @treturn boolean true if code is wrapped by Trapper, false otherwise
 function Trapper:isWrapped()
     if coroutine.running() then
         return true
     end
     return false
 end
 
---- Allow for overriding the Paused text and buttons on the abort
--- ConfirmBox displayed when InfoMessage dismissed in Trapper:UIFeedback()
-function Trapper:setPauseText(text, ok_text, cancel_text)
-    self.paused_text = text
-    self.paused_continue_text = ok_text
-    self.paused_abort_text = cancel_text
+--- Clear any left-over widget
+function Trapper:clear()
+    if self:isWrapped() then
+        if self.current_widget then
+            UIManager:close(self.current_widget)
+            UIManager:forceRePaint()
+            self.current_widget = nil
+        end
+    end
 end
 
---- Simple UI interaction via InfoMessage and ConfirmBox
---
--- Usage:
---
--- Trapper:UIFeedback("some text", true)
--- Trapper:UIFeedback("some text", true, "go on", "no more")
---   - display a ConfirmBox with the text and ok_text/cancel_text buttons.
---   - block and wait for user choice
---   - return the choice made: true if OK tapped, false if Cancel tapped or dismissed
---
--- Trapper:UIFeedback("some text")
--- Trapper:UIFeedback()
---   - display a InfoMessage with text
---     or keep existing InfoMessage if text = nil
---   - return true
---   UNLESS the previous widget was itself a InfoMessage and it has been
---   dismissed (by Tap on the screen), in which case:
---     - don't display the new InfoMessage
---     - return false
---   So, we can only know a InfoMessage has been dismissed when we
---   try to display a new one (we can't do better than that with coroutines).
---   So, don't hesitate to call it regularly (each call costs 100ms), between
---   steps of the work, to provide good responsiveness.
---   Trapper:UIFeedback() is a shortcut to get dismiss info while keeping
---   the existing InfoMessage displayed.
---
--- Trapper:UIFeedback(false)
---   Remove any remaining widget and consider the usage of Trapper done.
---
--- @text string text message, or nil/false
--- @confirmbox[opt=false] bool use ConfirmBox if true, otherwise use InfoMessage
--- @ok_text string text for ConfirmBox Ok button
--- @cancel_text string text for ConfirmBox Cancel button
--- @treturn bool true if ConfirmBox OK or InfoMessage not dismissed, false if Cancel or dismissed
-function Trapper:UIFeedback(text, show_confirmbox, ok_text, cancel_text)
-
-    -- We need to be called from an already created and running coroutine,
-    -- ie: Trapper:wrap() is executing the function calling us
+--- Clear any left-over widget and reset Trapper state
```

Comment:
Clears

widget or widgets? any goes with widgets; otherwise I'd just say "clears left-over widget" (in telegram style)